client/proxy/proxy.go

```diff
@@ -0,0 +1,147 @@
+package proxy
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"istio.io/manager/apiserver"
+	"istio.io/manager/model"
+)
+
+// ManagerClient is a client wrapper that contains the base URL and API version
+type ManagerClient struct {
+	base             url.URL
+	versionedAPIPath string
+	client           *http.Client
+}
+
+// NewManagerClient creates a new ManagerClient instance. It trims the apiVersion of leading and trailing slashes
+// and the base path of trailing slashes to ensure consistency
+func NewManagerClient(base url.URL, apiVersion string, client *http.Client) *ManagerClient {
+	base.Path = strings.TrimSuffix(base.Path, "/")
+	return &ManagerClient{
+		base:             base,
+		versionedAPIPath: strings.TrimPrefix(strings.TrimSuffix(apiVersion, "/"), "/"),
+		client:           client,
+	}
+}
+
+func (m *ManagerClient) do(request *http.Request) (*http.Response, error) {
+	fullURL, err := url.Parse(fmt.Sprintf("%s/%s/%s",
+		m.base.String(), m.versionedAPIPath, request.URL.String()))
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse URL: %v", err)
+	}
+	request.URL = fullURL
+	response, err := m.client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		return nil, fmt.Errorf("received non-success status code %v", response.StatusCode)
+	}
+	return response, nil
+}
+
+// GetConfig retrieves the configuration resource for the passed key
+func (m *ManagerClient) GetConfig(key model.Key) (config *apiserver.Config, err error) {
+	response, err := m.doConfigCRUD(key, http.MethodGet, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() { err = response.Body.Close() }()
```

Comment:
It can't because it fails the linting (for both suggestions)
`warning: Errors unhandled.,LOW,HIGH (gas) -> defer func() ( _ = response.Body.Close() }()`
`warning: error return value not checked (defer response.Body.Close()) (errcheck) -> defer response.Body.Close()`